src/native/libs/System.Native/pal_io.c

```diff
@@ -369,6 +369,50 @@ int32_t SystemNative_Unlink(const char* path)
     return result;
 }
 
+int32_t SystemNative_MemfdSupported(void)
+{
+#if HAVE_MEMFD_CREATE
+#ifdef TARGET_LINUX
+    struct utsname uts;
+    int32_t major, minor;
+
+    // memfd_create is only known to work properly on kernel version > 3.17 and throws SIGSEGV instead of ENOTSUP
```

Comment:
> and throws SIGSEGV instead of ENOTSUP

I'm not clear on what this comment is saying. Is it saying that <= 3.17 it uses SIGSEGV or after 3.17 it uses SIGSEGV?

Reply:
I haven't tested it on <3.17, just read https://github.com/dotnet/runtime/issues/92905#issuecomment-1744364761 and followed that pattern. In <3.17, calling `memfd_create` apparently fails with `SIGSEGV` (due to a possible kernel bug?).

Reply:
[According to documentation](https://man7.org/linux/man-pages/man2/memfd_create.2.html) `memfd_create` did not exist prior to kernel 3.17.

Reply:
memfd_create is a libc call, if you have older kernel, then libc function uses syscall() which is supposed to return ENOSYS, but it apparently throws sigsegv.

Reply:
> I haven't tested it on <3.17, just read [#92905 (comment)](https://github.com/dotnet/runtime/issues/92905#issuecomment-1744364761) and followed that pattern. In <3.17, calling `memfd_create` apparently fails with `SIGSEGV` (due to a possible kernel bug?).

Ok, can we tweak the comment to make that a tad more clear, e.g.
```suggestion
    // memfd_create is known to only work properly on kernel version > 3.17.
    // On earlier versions, it may raise SIGSEGV instead of returning ENOTSUP.
```